design/sketch.webidl

```diff
@@ -0,0 +1,532 @@
+};
+
+dictionary WebGPUTextureDescriptor {
+    u32 width;
+    u32 height;
+    u32 depth;
+    u32 arraySize;
+    WebGPUTextureDimensionEnum dimension;
+    WebGPUTextureFormatEnum format;
+    WebGPUTextureUsageFlags usage;
+};
+
+interface WebGPUTexture {
+    WebGPUTextureView createTextureView(WebGPUTextureViewDescriptor desc);
+};
+
+// Sampler
+typedef u32 WebGPUFilterModeEnum;
+interface WebGPUFilterMode {
+    const u32 NEAREST = 0;
+    const u32 LINEAR = 1;
+};
+
+dictionary WebGPUSamplerDescriptor {
+    WebGPUFilterModeEnum magFilter;
+    WebGPUFilterModeEnum minFilter;
+    WebGPUFilterModeEnum mipmapFilter;
+};
+
+interface WebGPUSampler {
+};
+
+// ****************************************************************************
+// BINDING MODEL (bindgroup layout, bindgroup)
+// ****************************************************************************
+
+// BindGroupLayout
+typedef u32 WebGPUShaderStageFlags;
+interface WebGPUShaderStageBit {
+    const u32 NONE = 0;
+    const u32 VERTEX = 1;
+    const u32 FRAGMENT = 2;
+    const u32 COMPUTE = 4;
+};
+
+typedef u32 WebGPUBindingTypeEnum;
+interface WebGPUBindingType {
+    const u32 UNIFORM_BUFFER = 0;
+    const u32 SAMPLER = 1;
+    const u32 SAMPLED_TEXTURE = 2;
+    const u32 STORAGE_BUFFER = 3;
+    // TODO other binding types (storage images, ...)
+};
+
+dictionary WebGPUBindGroupBinding {
+    WebGPUShaderStageFlags visibility;
+    WebGPUBindingType type;
+    u32 start;
+    u32 count;
+};
+
+dictionary WebGPUBindGroupLayoutDescriptor {
+    sequence<WebGPUBindGroupBinding> bindingTypes;
+};
+
+interface WebGPUBindGroupLayout {
+};
+
+// PipelineLayout
+dictionary WebGPUPipelineLayoutDescriptor {
+    sequence<WebGPUBindGroupLayout> bindGroupLayouts;
+};
+
+interface WebGPUPipelineLayout {
+};
+
+// BindGroup
+dictionary WebGPUBufferBinding {
+    WebGPUBuffer buffer;
+    u32 offset;
+    u32 size;
+};
+
+typedef (WebGPUSampler or WebGPUTextureView or WebGPUBufferBinding) WebGPUBindingResource;
+
+dictionary WebGPUBinding {
+    sequence<WebGPUBindingResource> resources;
+    u32 start;
+    u32 count;
+};
+
+dictionary WebGPUBindGroupDescriptor {
+    WebGPUBindGroupLayout layout;
+    sequence<WebGPUBinding> bindings; 
+};
+
+interface WebGPUBindGroup {
+};
+
+// ****************************************************************************
+// PIPELINE CREATION (blend state, DS state, ..., pipelines)
+// ****************************************************************************
+
+// BlendState
+typedef u32 WebGPUBlendFactorEnum;
+interface WebGPUBlendFactor {
+    const u32 ZERO = 0;
+    const u32 ONE = 1;
+    const u32 SRC_COLOR = 2;
+    const u32 ONE_MINUS_SRC_COLOR = 3;
+    const u32 SRC_ALPHA = 4;
+    const u32 ONE_MINUS_SRC_ALPHA = 5;
+    const u32 DST_COLOR = 6;
+    const u32 ONE_MINUS_DST_COLOR = 7;
+    const u32 DST_ALPHA = 8;
+    const u32 ONE_MINUS_DST_ALPHA = 9;
+    const u32 SRC_ALPHA_SATURATED = 10;
+    const u32 BLEND_COLOR = 11;
```

Comment:
OpenGL, Vulkan, Metal (and I assume D3D12 too)  has separate enum for constant alpha. At glance it seems that only `BLEND_COLOR` isn't enough since one can use `BLEND_ALPHA` as blending factor for rgb to get `blendA, blendA, blendA` factor.

Reply:
I didn't see one in D3D12